jenkins_autojobs/git.py

```diff
@@ -96,15 +96,18 @@ def create_job(ref, template, config, ref_config):
 
     # Set branch.
     el = scm_el.xpath('//hudson.plugins.git.BranchSpec/name')[0]
-    # :todo: jenkins is being very capricious about the branch-spec
-    # el.text = '%s/%s' % (remote, shortref)  # :todo:
-    el.text = shortref
 
-    # Set the branch that the git plugin will locally checkout to.
-    el = scm_el.xpath('//localBranch')
-    el = lxml.etree.SubElement(scm_el, 'localBranch') if not el else el[0]
-
-    el.text = shortref  # the original shortref (with '/')
+    # If el.text is not empty, the template has already had a treeish specified
+    # do not erase it
+    if not el.text:
+	# :todo: jenkins is being very capricious about the branch-spec
```

Comment:
It can't be seen in the github diff viewer, but these lines start with tabs instead of spaces. This would result in a `TabError` in Python 3.